### ISSUE TITLE
Get/maintain Cactus conda package

### ISSUE THREAD

**Issue by glennhickey** [ComparativeGenomicsToolkit/cactus#910]
This would help a lot of people who seem to have conflicts between their conda environments and the virtualenv-based Cactus installation.  Apparently there is some prior work, [see this comment](https://github.com/ComparativeGenomicsToolkit/hal/issues/262#issuecomment-1387534845)

**Comment by amizeranschi** [comment]
Hi @glennhickey 

Are you still taking this idea into consideration for future releases? It would be great to have updated versions of Cactus and HAL in Bioconda.

**Comment by glennhickey** [comment]
Yeah, for sure.  Unfortunately I don't personally have any time for at least a few weeks to look at it -- am keeping an eye out for volunteers though.

**Comment by twrightsman** [comment]
I've been looking into and starting to chip away at what is needed to get Cactus packaged in Bioconda. I have been trying to approach this using dynamic linking and conda-forge/Bioconda-provided dependencies rather than vendored ones. To the best of my knowledge, the dependencies for Cactus v2.7.1 and their current Conda packaging status are:

## Build (Conda `build:` dependencies)

- [x] `compiler('c')`
- [x] `make`
- [x] `git`
    - is this optional? cloned repo is not available in Conda build, only tarball; version is provided as an env var
- [x] `pkg-config`
- [x] `cmake`

## Target architecture-specific libraries (Conda `host:` dependencies)

- [x] `python` (3.7+)
- [x] `zlib`
- [x] `bzip2`
- [x] `hdf5`
- [x] `lzo`
- [x] `tokyocabinet`
- [x] `xz` (lzma)
- [x] `libxml2`
- [x] `openssl`
- [x] `libpng`
- [x] `libuuid`
- [x] `libcurl`
- [x] `libffi`
- [x] `libdeflate`
- [x] `jemalloc`
- [x] `pip`
- [x] `abpoa` (1.5.1)
- [ ] [`sonlib`](https://github.com/ComparativeGenomicsToolkit/sonLib) (2.0?)
    - updated packaging in my `bioconda-recipes` [fork](https://github.com/twrightsman/bioconda-recipes/tree/update-sonlib)
    - needs tagged version: https://github.com/ComparativeGenomicsToolkit/sonLib/issues/23
    - `sonLibTests` currently [segfaults](https://github.com/bioconda/bioconda-recipes/compare/master...twrightsman:bioconda-recipes:update-sonlib#diff-c539f7198f4c622561d92c24307759600c1a71deea22c377c51a04bfaf0336b0R18-R19)
    - [ ] `cutest`: https://github.com/conda-forge/staged-recipes/pull/25087
    - [ ] `quicktree`: https://github.com/bioconda/bioconda-recipes/pull/45326
- [ ] [`matchingAndOrdering`](https://github.com/ComparativeGenomicsToolkit/matchingAndOrdering)
    - needs tagged version
- [ ] [`cpecan`](https://github.com/ComparativeGenomicsToolkit/cPecan)
    - needs tagged version
    - binaries needed at runtime too?
- [ ] [`pinchesandcacti`](https://github.com/ComparativeGenomicsToolkit/pinchesAndCacti)
    - needs tagged version

## Runtime (Conda `run:` dependencies)

- [x] `python` (3.7+)
- [x] `toil` (6.0.0)
- [x] `lastz`
- [ ] [`hal`](https://github.com/ComparativeGenomicsToolkit/hal) (2.3): https://github.com/bioconda/bioconda-recipes/pull/45295
    - needed at build time?
    - kent tools?
    - [x] `phast`
- [ ] [`paffy`](https://github.com/ComparativeGenomicsToolkit/paffy)
    - needs tagged version
    - needed at build time?

**Comment by twrightsman** [comment]
I don't have time to work on this anymore; the devendoring/shlib builds take a while to figure out. Anyone interested in picking this up can start with finishing up the hal packaging by adding the unit tests to the test phase of conda-build, see my hal [fork](https://github.com/twrightsman/hal/tree/add-devendored-shlib-build) and https://github.com/bioconda/bioconda-recipes/pull/45295.

**Comment by glennhickey** [comment]
Thanks for your efforts.  I haven't had time to do this either, but yeah, it'd be great if there are any conda / cactus users that want to contribute.
